packages/core/src/model.types.ts

```diff
@@ -1,3 +1,4 @@
+import { ExtraGenerics } from '.';
```

Comment:
could you sum up why you have chosen this approach and how do you imagine this evolving in the future?